spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/TaskExecutionDao.java

```diff
@@ -201,4 +203,32 @@ TaskExecution startTaskExecution(long executionId, String taskName,
 	 */
 	void updateExternalExecutionId(long taskExecutionId,
 			String externalExecutionId);
+
+	/**
+	 * Returns a {@link List} of the latest {@link TaskExecution} for 1 or more task names.
+	 *
+	 * Latest is defined by the most recent start time. A {@link TaskExecution} does not have to be finished
+	 * (The results may including pending {@link TaskExecution}s).
+	 *
+	 * It is theoretically possible that a {@link TaskExecution} with the same name to have more than 1
+	 * {@link TaskExecution} for the exact same start time. In that case the {@link TaskExecution} with the
+	 * highest Task Execution ID is returned.
+	 *
+	 * This method will not consider end times in its calculations. Thus, when a task execution {@code A} starts
+	 * after task execution {@code B} but finishes BEFORE task execution {@code A}, then task execution {@code B}
+	 * is being returned.
+	 *
+	 * @param taskNames At least 1 task name must be provided
+	 * @return List of TaskExecutions. May be empty but never null.
+	 */
+	List<TaskExecution> getLatestTaskExecutionsByTaskNames(String... taskNames);
+
+	/**
+	 * Returns the latest task execution for a given task name.
```

Comment:
Is the algorithm used for choosing the the "latest" the same as for `getLatestTaskExecutionsByTaskNames`?  That's unclear from the javadoc.